tests/detectors/test_detectors_fileformats.py

```diff
@@ -0,0 +1,141 @@
+    d = garak.detectors.fileformats.FileIsPickled()
+    nonpickle_file_attempt = Attempt(prompt="")
+    nonpickle_file_attempt.notes["format"] = "local filename"
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as nonpickle_file:
+        nonpickle_file.write(
+            "Do you know what the sad part is, Odo? I'm a very good tailor."
+        )
+        nonpickle_file.close()
+        nonpickle_file_attempt.outputs = [nonpickle_file.name]
+        l = list(d.detect(nonpickle_file_attempt))
+        assert l == [0.0]
+
+
+def test_fileispickled_pickle_default():
+    d = garak.detectors.fileformats.FileIsPickled()
+    pickle_file_attempt = Attempt(prompt="")
+    pickle_file_attempt.notes["format"] = "local filename"
+    with tempfile.NamedTemporaryFile(mode="wb+", delete=False) as pickle_file:
+        test_data_struct = {"key": ["value", 37176]}
+        pickle.dump(test_data_struct, pickle_file)
+        pickle_file.close()
+        pickle_file_attempt.outputs = [pickle_file.name]
+        l = list(d.detect(pickle_file_attempt))
+        assert l == [1.0]
+
+
+pickle_versions = [0, 1, 2, 3, 4, 5]
+
+
+@pytest.mark.parametrize("pickle_ver", pickle_versions)
+def test_fileispickled_pickle_versioned(pickle_ver):
+    d = garak.detectors.fileformats.FileIsPickled()
+    pickle_file_attempt = Attempt(prompt="")
+    pickle_file_attempt.notes["format"] = "local filename"
+    with tempfile.NamedTemporaryFile(mode="wb+", delete=False) as pickle_file:
+        test_data_struct = {"key": ["value", 37176]}
+        pickle.Pickler(pickle_file, pickle_ver).dump(test_data_struct)
+        pickle_file.close()
+        pickle_file_attempt.outputs = [pickle_file.name]
+        l = list(d.detect(pickle_file_attempt))
+        assert l == [1.0]
+
+
+def test_fileisexectuable_nonexec():
+    exec_attempt = garak.attempt.Attempt(prompt="")
+    exec_attempt.notes["format"] = "local filename"
+    d = garak.detectors.fileformats.FileIsExecutable()
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as test_file:
+        test_file.write("I'm a very good tailor.")
+        test_file.close()
+        exec_attempt.outputs = [test_file.name]
+        results = d.detect(exec_attempt)
+        assert list(results) == [0.0]
+
+
+path_prefix = Path("tests/resources/fileformats/exec_files/")
+
+
+@pytest.fixture
+def decoded_filename(request, encoded_exec_filename):
+    with open(encoded_exec_filename, "r", encoding="utf-8") as encodedfile:
+        with tempfile.NamedTemporaryFile(mode="wb+", delete=False) as binfile:
+            binfile.write(binascii.a2b_base64(encodedfile.read()))
+            binfile.flush()
+            yield binfile.name
+
+    def remove_decoded():
+        with contextlib.suppress(FileNotFoundError):
+            Path(binfile.name).unlink()
+
+    request.addfinalizer(remove_decoded)
+
+
+@pytest.mark.parametrize(
+    "encoded_exec_filename", glob.glob(str(path_prefix / "*.base64"))
+)
```

Comment:
May be a good idea to skip these test when `magic` is not available, not all development environments will need to interact with this probe, contributors will be more likely to run tests if they don't need to hit edge cases to get the suite to run at all.

```suggestion
try:
    import magic
except ImportError as e:
    magic == None


@pytest.mark.skipif(magic is None, reason="libmagic not available")
@pytest.mark.parametrize(
    "encoded_exec_filename", glob.glob(str(path_prefix / "*.base64"))
)
```

Reply:
yes, this makes sense, good catch